ui/apps/platform/src/Containers/Collections/CollectionResults.tsx

```diff
@@ -1,7 +1,153 @@
-import React from 'react';
+import React, { useCallback, useEffect, useState, useMemo } from 'react';
+import {
+    Button,
+    Divider,
+    EmptyState,
+    EmptyStateIcon,
+    EmptyStateVariant,
+    Flex,
+    FlexItem,
+    Select,
+    SearchInput,
+    SelectOption,
+    debounce,
+} from '@patternfly/react-core';
+import { ListIcon } from '@patternfly/react-icons';
+import useSelectToggle from 'hooks/patternfly/useSelectToggle';
+import ResourceIcon from 'Components/PatternFly/ResourceIcon';
 
-function CollectionResults() {
-    return <></>;
+import { dryRunCollection } from 'services/CollectionsService';
+import { ListDeployment } from 'types/deployment.proto';
+
+function CollectionResults({ dryRunConfig }) {
+    const { isOpen, onToggle, closeSelect } = useSelectToggle();
+    const [selected, setSelected] = useState('Deployment');
+    const [filterText, setFilterText] = useState('');
+    const [isEndOfResults, setIsEndOfResults] = useState(false);
+    const [deployments, setDeployments] = useState<ListDeployment[]>([]);
+    const [hasRuleSelected, setHasRuleSelected] = useState(false);
+
+    function onRuleOptionSelect(_, value) {
+        setSelected(value);
+        setFilterText('');
+        closeSelect();
+    }
+
+    const fetchDryRun = useCallback(
+        (page: number) => {
+            const pageSize = 10;
+            const query = filterText ? `${selected}:${filterText}` : '';
+            const { request } = dryRunCollection(dryRunConfig, query, page, pageSize);
+            request
+                .then((results) => {
+                    setIsEndOfResults(results.length < 10);
+                    setDeployments((current) => (page === 0 ? results : [...current, ...results]));
+                    setHasRuleSelected(true);
+                })
+                .catch(() => {
+                    // TODO: indicate results not loading properly?
+                });
+        },
+        [dryRunConfig, filterText, selected]
+    );
+
+    useEffect(() => {
+        if (dryRunConfig.resourceSelectors?.[0]?.rules?.length > 0) {
```

Comment:
We also need to account for the case where there are no rules, but the user has attached a collection. I would say that a collection is considered empty if:
- it has no `resourceSelectors`
or
- Every `resourceSelector` has zero rules _and_ the embedded collection array is empty

Reply:
Also regarding the above, you should be able to calculate this right in the component's render call and remove the need to track `hasRuleSelected` in state.